codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -379,21 +379,36 @@ public T add(Headers<? extends K, ? extends V, ?> headers) {
         if (headers == this) {
             throw new IllegalArgumentException("can't add to itself.");
         }
+        addImpl(headers);
+        return thisT();
+    }
+
+    protected void addImpl(Headers<? extends K, ? extends V, ?> headers) {
         if (headers instanceof DefaultHeaders) {
             @SuppressWarnings("unchecked")
-            DefaultHeaders<K, V, T> defaultHeaders = (DefaultHeaders<K, V, T>) headers;
-            HeaderEntry<K, V> e = defaultHeaders.head.after;
-            while (e != defaultHeaders.head) {
-                add(e.key, e.value);
-                e = e.after;
+            DefaultHeaders<? extends K, ? extends V, T> defaultHeaders =
+                    (DefaultHeaders<? extends K, ? extends V, T>) headers;
+            HeaderEntry<? extends K, ? extends V> e = defaultHeaders.head.after;
+            if (defaultHeaders.hashingStrategy == this.hashingStrategy &&
+                    defaultHeaders.nameValidator == this.nameValidator) {
+                // Fastest copy
+                while (e != defaultHeaders.head) {
+                    add0(e.hash, index(e.hash), e.key, e.value);
+                    e = e.after;
+                }
+            } else {
+                // Fast copy
```

Comment:
You have 2 `http2AddAllFastest` labels and 1 `http2AddAllFast` in the text above. I don't see `http2AddAllSlow`. Can you clarify the benchmark names and association with numbers?

I'm guessing the numbers follow the `Fastest`, `Fast`, `Slow` terminology?